flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/factories/StreamTableSinkFactory.scala

```diff
@@ -20,15 +20,15 @@ package org.apache.flink.table.factories
 
 import java.util
 
-import org.apache.flink.table.sinks.StreamTableSink
+import org.apache.flink.table.sinks.{StreamTableSink, TableSink}
 
 /**
   * A factory to create configured table sink instances in a streaming environment based on
-  * string-based properties. See also [[TableFactory]] for more information.
+  * string-based properties. See also [[TableSinkFactory]] for more information.
   *
   * @tparam T type of records that the factory consumes
   */
-trait StreamTableSinkFactory[T] extends TableFactory {
+trait StreamTableSinkFactory[T] extends TableSinkFactory[T] {
```

Comment:
Can we also convert this interface to Java now and move them to its final destination?

Reply:
I'm not sure. If we put this class into the java-bridge, the scala-bridge needs to depend on java-bridge, because java users also need this class.
Do you think it's right to make scala-bridge depends on java-bridge?

Reply:
We had this discussions in the past when creating FLIP-32. I cite the important parts:
"Currently, we cannot add interfaces for connectors into the common module as classes such as `StreamTableSource` or `BatchTableSource` require a dependency to `DataStream` or `DataSet` API classes. This might change in the future once we reworked our TableSource and TableSink interfaces. For now, extension points for connectors are located in `flink-table-api-*-bridge` and tightly integrate with the target API."

`StreamTableSource` and sink use the Java version of the DataStream API, thus, those two interfaces belong into a Java bridge. If users want to implement a Java source and use the Scala Table API, they can simply add both modules. This is not a problem because classes in both bridging modules are prefixed with `api.java` or `api.scala` or do not have class name clashes.

In the near future we will deprecate those interfaces and have just one unified interface in `common` that does not depend in DataStream API anymore but returns a `SourceFunction`.

Reply:
The `SourceFunction` you mentioned is the new source interface of "FLIP-27: Refactor Source Interface" ? Because the current `SourceFunction` is still in `flink-streaming-java` module.

Reply:
@twalthr Thanks for the detailed explanation. I find that scala users need to depend on java-bridge any way either we put this class in planner or in java-bridge. I will port it into the java-bridge. Thank you.

Reply:
@wuchong yes I'm talking about FLIP-27. The new unified `SourceFunction` should be in `flink-core` close to `MapFunction` etc.